okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -493,14 +493,21 @@ public void close() {}
           if (proxy == null) {
             sock = new Socket(address.getAddress(), address.getPort());
           } else {
-            sock = createHttpProxySocket(
-                address,
-                proxy.getProxyAddress(),
-                proxy.getUsername(),
-                proxy.getPassword()
-            );
+            if (proxy.getProxyAddress() == null) {
+              // This should never happen, case added for readability
+              throw new AssertionError();
+            } else if (proxy.getProxyAddress() instanceof InetSocketAddress) {
+              sock = createHttpProxySocket(
+                  address,
+                  (InetSocketAddress)proxy.getProxyAddress(),
+                  proxy.getUsername(),
+                  proxy.getPassword()
+              );
+            } else {
```

Comment:
Make this an if above the createHttpProxySocket, that way there's no need for the else:

```
if (!(proxy.getProxyAddress() instanceof InetSocketAddress)) {
  throw ...;
}
sock = createHttpProxySocket(...):
```

Reply:
added these last changes @ejona86